src/html_json.js

```diff
@@ -181,12 +157,34 @@ module.exports.main = async (context, action) => {
   const {
     owner, repo, ref, path,
   } = action.request.params;
+
   const { logger } = action;
 
+  const secrets = setdefault(action, 'secrets', {});
+  secrets.REPO_RAW_ROOT = secrets.REPO_RAW_ROOT || 'https://raw.githubusercontent.com/';
+  secrets.HTTP_TIMEOUT = secrets.HTTP_TIMEOUT || 10000;
+
+  const loadConfig = async () => {
+    const downloader = new Downloader(context, action, {
+      forceHttp1: owner === 'helix', // use http1 for tests
+    });
+
+    const indexYAML = await downloader.fetchGithub({
+      owner, repo, ref, path: '/helix-query.yaml',
+    });
+    if (indexYAML.status !== 200) {
+      logger.warn(`Unable to fetch helix-query.yaml: ${indexYAML.status}`);
+      return {
+        indices: [],
+      };
+    }
+    return (await new IndexConfig()
+      .withSource(indexYAML.body)
+      .init()).toJSON();
+  };
+
+  const config = await loadConfig();
```

Comment:
Using a function *inside* a function, to be able to use the arguments in the outer scope, not sure, whether this is discouraged, though.

Reply:
I think this is ok.